### PR TITLE
Ensure 'fetched' dir exists before use

### PR DESCRIPTION
**Pull request by abays**
Potential fix for https://github.com/redhat-partner-solutions/crucible/issues/95

### CODE REVIEW

File: roles/generate_ssh_key_pair/tasks/main.yml [hunk 1]
```diff
@@ -4,6 +4,12 @@
     path: "{{ key_pair_dir }}"
     state: absent
 
+- name: "Make sure {{ fetched_dest }} exists"
+  file:
+    path: "{{ fetched_dest }}"
+    state: directory
+    recurse: yes
```
Comment (nocturnalastro) [inline roles/generate_ssh_key_pair/tasks/main.yml:11]: Hmm we've been a bit fast an lose with `bastion` and `localhost` throughout the playbooks. Can we still delegate this to `localhost`. While we strongly suggest they should be the same there is no guarantee they are. I'll take a note to check the playbooks over to check that the distinction is correct and write up a contributor note to explain the difference.

Comment (abays) [reply roles/generate_ssh_key_pair/tasks/main.yml:11]: Ack, will add the delegate

**Review by snecklifter — APPROVED** [review]
LGTM, I have also hit that from memory.

File: roles/generate_ssh_key_pair/tasks/main.yml (hunk 1 above)
Comment (arjuhe) [reply roles/generate_ssh_key_pair/tasks/main.yml:11]: @abays thanks for adding that. Coincidentally I brought up the bastion/localhost thing in a separate PR earlier today. 😄

**Review by arjuhe — CHANGES_REQUESTED** [review]
Just add the delegate and we are good to go.

**Review by nocturnalastro — APPROVED** [review]

**Pull request merged**